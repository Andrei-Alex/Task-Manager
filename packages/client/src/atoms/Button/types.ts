import { CustomStyles } from "../../utils/types/styles";

export interface IButton {
  text: string;
  onClick?: () => void;
  width?: number;
  type: "submit" | "button" | "reset" | undefined;
  containerStyle?: CustomStyles;
  variant?:
    | "outline"
    | "white"
    | "light"
    | "default"
    | "filled"
    | "gradient"
    | "subtle";
  loading?: boolean;
  disabled?: boolean;
  size?: "xs" | "sm" | "md" | "lg" | "xl";
  color?: string;
  gradient?: { from: string; to: string };
  fullWidth?: boolean;
}
