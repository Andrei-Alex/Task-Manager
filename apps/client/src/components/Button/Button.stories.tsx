import type { ComponentStory, ComponentMeta } from '@storybook/react';
import React from 'react';
import Button from './Button';

const Story: ComponentMeta<typeof Button> = {
  component: Button,
  title: 'Button',
  argTypes: {
    text: {
      name: 'Button text',
      type: {
        name: 'string',
        required: false,
      },
      defaultValue: 'Click',
      description: 'Display text',
      control: {
        type: 'text',
      },
    },
    background: {
      name: 'Background color',
      type: {
        name: 'string',
        required: false,
      },
      defaultValue: 'green',
      description: 'Background color',
      control: {
        type: 'text',
      },
    },
    color: {
      name: 'Text color',
      type: {
        name: 'string',
        required: false,
      },
      defaultValue: 'white',
      description: 'Text color',
      control: {
        type: 'text',
      },
    },
    additionalText: {
      name: 'Additional text',
      type: {
        name: 'string',
        required: false,
      },
      defaultValue: false,
      description: 'Display additional Text',
      control: {
        type: 'boolean',
      },
    },
  },
};
export default Story;

const Template: ComponentStory<typeof Button> = args => <Button {...args} />;

export const ClickButton = Template.bind({});
ClickButton.args = {
  text: 'Click',
  background: 'green',
  color: 'white',
  additionalText: false,
};
