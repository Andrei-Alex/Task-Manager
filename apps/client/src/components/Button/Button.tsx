import { styles, IButton } from '.';

/**
 * Simple Button.
 *
 * ## Usage
 * ```jsx
 * <div>
 *  <p>Text AdditionalText</p>
 * </div>
 * ```
 * @param {string} text - Button text
 * @param {boolean} additionalText - Toogle Display text
 * @param {string} background - Background color
 * @param {string} color - Text color
 * @return {JSX} Display text in JSX elements
 */
export const Button = ({
  text,
  additionalText = false,
  background = 'green',
  color,
}: IButton) => {
  return (
    <div
      style={{ backgroundColor: background }}
      className={styles['container']}
    >
      <p style={{ color: color }}>
        {text}
        {additionalText ? ' Here !' : ''}
      </p>
    </div>
  );
};

export default Button;
