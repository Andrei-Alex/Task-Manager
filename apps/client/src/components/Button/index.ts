export { default as styles } from './Button.module.scss';
export type { IButton } from './types';
