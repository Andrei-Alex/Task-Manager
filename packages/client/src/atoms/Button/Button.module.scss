@import '../../styles/variables';

.container {
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 10px;
  height: 40px;
  font-size: 1.2rem;
  font-weight: 700;
  color: white;
  background-color: $mediumGray;
  font-family: $fontPrimary;
  border: none;
}
