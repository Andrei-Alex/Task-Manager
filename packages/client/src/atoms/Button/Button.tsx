import React from "react";
import { styles, IButton } from "./index";

/**
 * Simple Button.
 *
 * ## Usage
 * ```jsx
 * <button>
 *  text
 * <button>
 * ```
 * @param {string} text - Button text
 * @param {number} width - Button Width
 * @param {string} backgroundColor - Background color
 * @param {string} textColor - Text color
 * @param {function} callback - Callback function
 * @return {JSX} Display Button
 */
export const Button: React.FC<IButton> = ({
  width = "300",
  text,
  backgroundColor = "#828FA3",
  textColor,
  callback = () => null,
}) => {
  return (
    <button
      onClick={() => callback()}
      className={styles.container}
      style={{
        width: `${width}px`,
        backgroundColor: backgroundColor,
        color: textColor,
      }}
    >
      {text}
    </button>
  );
};

export default Button;
