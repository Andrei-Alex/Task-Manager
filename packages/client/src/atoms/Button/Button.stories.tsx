import type { Meta, StoryObj } from "@storybook/react";
import React from "react";
import Button from "./Button";

const meta: Meta<typeof Button> = {
  component: Button,
  title: "Atoms/Button",
  tags: ["autodocs"],
  argTypes: {
    text: {
      name: "Button text",
      type: {
        name: "string",
        required: false,
      },
      defaultValue: "Click",
      description: "Display text",
      control: {
        type: "text",
      },
    },
    backgroundColor: {
      name: "Background color",
      type: {
        name: "string",
        required: false,
      },
      defaultValue: "#828FA3",
      description: "Background color",
      control: {
        type: "text",
      },
    },
    textColor: {
      name: "Text color",
      type: {
        name: "string",
        required: false,
      },
      defaultValue: "white",
      description: "Text color",
      control: {
        type: "text",
      },
    },
    onClick: {
      name: "onClick",
      type: {
        name: "function",
        required: false,
      },
      description: "onClick callback function",
    },
  },
};
export default meta;
type Story = StoryObj<typeof Button>;

export const Default: Story = {
  args: {
    text: "Click me",
    textColor: "white",
  },
};
