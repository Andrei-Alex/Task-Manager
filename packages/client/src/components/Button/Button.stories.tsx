import type { Meta, StoryObj } from "@storybook/react";
import React from "react";
import Button from "./Button";

const meta: Meta<typeof Button> = {
  component: Button,
  title: "Button",
  tags: ["autodocs"],
  argTypes: {
    text: {
      name: "Button text",
      type: {
        name: "string",
        required: false,
      },
      defaultValue: "Click",
      description: "Display text",
      control: {
        type: "text",
      },
    },
    background: {
      name: "Background color",
      type: {
        name: "string",
        required: false,
      },
      defaultValue: "green",
      description: "Background color",
      control: {
        type: "text",
      },
    },
    color: {
      name: "Text color",
      type: {
        name: "string",
        required: false,
      },
      defaultValue: "white",
      description: "Text color",
      control: {
        type: "text",
      },
    },
    additionalText: {
      name: "Additional text",
      type: {
        name: "string",
        required: false,
      },
      defaultValue: false,
      description: "Display additional Text",
      control: {
        type: "boolean",
      },
    },
  },
};
export default meta;
type Story = StoryObj<typeof Button>;

export const Primary: Story = {
  args: {
    text: "Click",
    background: "green",
    color: "white",
    additionalText: false,
  },
};
