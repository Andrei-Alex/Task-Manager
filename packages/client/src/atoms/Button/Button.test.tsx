import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import Button from "./Button";
import renderer from "react-test-renderer";

const handleClick = jest.fn();
describe("Button", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Button text={"Click here"} buttonType={"button"} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should display Click here", () => {
    render(<Button text={"Click here"} buttonType={"button"} />);
    const button = screen.getByText("Click here");
    expect(button).toBeInTheDocument();
  });
  it("should have container class", () => {
    render(<Button text={"Click here"} buttonType={"button"} />);
    const button = screen.getByText("Click here");
    expect(button).toHaveClass(`button`);
  });
  it("should display Click here", () => {
    render(<Button text={"Click here"} buttonType={"button"} />);
    const button = screen.getByText("Click here");
    expect(button).toBeInTheDocument();
  });
  it("should call onClick function", async () => {
    render(
      <Button text={"Click here"} onClick={handleClick} buttonType={"button"} />
    );
    const button = screen.getByText("Click here");
    fireEvent.click(button);
    expect(handleClick).toBeCalledTimes(1);
  });
});
