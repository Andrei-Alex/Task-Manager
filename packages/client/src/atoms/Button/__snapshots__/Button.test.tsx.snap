// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div
  style={Object {}}
>
  <button
    className="button"
    onClick={[Function]}
    style={
      Object {
        "background": undefined,
        "color": undefined,
        "width": "100%",
      }
    }
    type="button"
  >
    Click here
  </button>
</div>
`;
